src/luxonis_ml/loader/augmentations.py

```diff
@@ -334,4 +389,197 @@ def __init__(
+
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.border_value = border_value
+        self.mask_value = mask_value
+
+    def update_params(self, params: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        params = super().update_params(params, **kwargs)
+
+        img_height = params["rows"]
+        img_width = params["cols"]
+
+        ratio = min(self.height / img_height, self.width / img_width)
+        new_height = int(img_height * ratio)
+        new_width = int(img_width * ratio)
+
+        # only supports center alignment
+        pad_top = (self.height - new_height) // 2
+        pad_bottom = pad_top
+
+        pad_left = (self.width - new_width) // 2
+        pad_right = pad_left
+
+        params.update(
+            {
+                "pad_top": pad_top,
+                "pad_bottom": pad_bottom,
+                "pad_left": pad_left,
+                "pad_right": pad_right,
+            }
+        )
+
+        return params
+
+    def apply(
+        self,
+        img: np.ndarray,
+        pad_top: int,
+        pad_bottom: int,
+        pad_left: int,
+        pad_right: int,
+        **params
+    ) -> np.ndarray:
+        resized_img = cv2.resize(
+            img,
+            (self.width - pad_left - pad_right, self.height - pad_top - pad_bottom),
+            interpolation=self.interpolation,
+        )
+        img_out = cv2.copyMakeBorder(
+            resized_img,
+            pad_top,
+            pad_bottom,
+            pad_left,
+            pad_right,
+            cv2.BORDER_CONSTANT,
+            self.border_value,
+        )
+        img_out = img_out.astype(img.dtype)
+        return img_out
+
+    def apply_to_mask(
+        self,
+        img: np.ndarray,
+        pad_top: int,
+        pad_bottom: int,
+        pad_left: int,
+        pad_right: int,
+        **params
+    ) -> np.ndarray:
+        resized_img = cv2.resize(
+            img,
+            (self.width - pad_left - pad_right, self.height - pad_top - pad_bottom),
+            interpolation=cv2.INTER_NEAREST,
+        )
+        img_out = cv2.copyMakeBorder(
+            resized_img,
+            pad_top,
+            pad_bottom,
+            pad_left,
+            pad_right,
+            cv2.BORDER_CONSTANT,
+            self.mask_value,
+        )
+        img_out = img_out.astype(img.dtype)
+        return img_out
+
+    def apply_to_bbox(
+        self,
+        bbox: BoxInternalType,
+        pad_top: int,
+        pad_bottom: int,
+        pad_left: int,
+        pad_right: int,
+        **params
+    ) -> BoxInternalType:
+        x_min, y_min, x_max, y_max = denormalize_bbox(
+            bbox, self.height - pad_top - pad_bottom, self.width - pad_left - pad_right
+        )[:4]
+        bbox = [x_min + pad_left, y_min + pad_top, x_max + pad_left, y_max + pad_top]
+        # clip bbox to image, ignoring padding
+        for i in range(4):
```

Comment:
I assume albumentations expects list of floats and not a numpy array?
Otherwise I think np.clip(..., min=[pad_left, pad_top, pad_left, pad_top], max=...) should be faster.